streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -62,206 +64,229 @@
-        );
+        collector.send(topic, "3", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "9", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "27", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "81", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "243", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "28", "0", headers, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "82", "0", headers, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "244", "0", headers, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.send(topic, "245", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
 
-        final Headers headers = new RecordHeaders(new Header[]{new RecordHeader("key", "value".getBytes())});
+        final Map<TopicPartition, Long> offsets = collector.offsets();
 
-        collector.send("topic1", "3", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
-        collector.send("topic1", "9", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
-        collector.send("topic1", "27", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
-        collector.send("topic1", "81", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
-        collector.send("topic1", "243", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+        assertEquals(4L, (long) offsets.get(new TopicPartition(topic, 0)));
+        assertEquals(2L, (long) offsets.get(new TopicPartition(topic, 1)));
+        assertEquals(0L, (long) offsets.get(new TopicPartition(topic, 2)));
+        assertEquals(9, producer.history().size());
 
-        collector.send("topic1", "28", "0", headers, null, stringSerializer, stringSerializer, streamPartitioner);
-        collector.send("topic1", "82", "0", headers, null, stringSerializer, stringSerializer, streamPartitioner);
-        collector.send("topic1", "244", "0", headers, null, stringSerializer, stringSerializer, streamPartitioner);
+        // returned offsets should not be modified
+        final TopicPartition topicPartition = new TopicPartition(topic, 0);
+        assertThrows(UnsupportedOperationException.class, () -> offsets.put(topicPartition, 50L));
+    }
 
-        collector.send("topic1", "245", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+    @Test
+    public void shouldSendWithNoPartition() {
+        final Headers headers = new RecordHeaders(new Header[]{new RecordHeader("key", "value".getBytes())});
```

Comment:
Why do we need to pass headers in this test?

Reply:
No we do not necessarily need to, do you have any concerns about this?